### PR TITLE
Upgrade rails

### PR DESCRIPTION
**Pull request by whoojemaflip**
Upgrades Rails from `4.2.7.1` -> `5.0.2`, using the [recommended approach](http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-4-2-to-rails-5-0).

Includes some style changes, dependency upgrades and fixes for console warnings.

[Trello](https://trello.com/c/UvadVLm8/72-upgrade-dependencies-on-our-applications-including-rails)

### CODE REVIEW

File: app/controllers/taggings_controller.rb [hunk 1]
```diff
@@ -55,12 +58,17 @@ def update
...
+    )
   end
 
 private
 
   def lookup_params
-    params[:content_lookup_form] || { base_path: "/#{params[:slug]}" }
+    params.require(:content_lookup_form).permit(:base_path)
+  rescue ActionController::ParameterMissing
+    { base_path: "/#{params[:slug]}" }
```
Comment (andrewgarner) [inline app/controllers/taggings_controller.rb]: We don't really require the `content_lookup_form` param so maybe we could do this?

```rb
params
  .fetch(:content_lookup_form, { base_path: "/#{params[:slug]}" })
  .permit(:base_path)
```

File: app/controllers/taxons_controller.rb [hunk 2]
```diff
@@ -108,6 +108,19 @@ def discard_draft
 
 private
 
+  def permitted_params
```
Comment (andrewgarner) [inline app/controllers/taxons_controller.rb]: The Rails convention would be to call this method `taxon_params`.

File: config/application.rb [hunk 3]
```diff
@@ -12,9 +12,5 @@
 
 module ContentTagger
   class Application < Rails::Application
-    config.active_record.raise_in_transactional_callbacks = true
-    config.eager_load_paths += %W(#{config.root}/lib)
-
-    config.action_view.field_error_proc = proc { |html_tag, _| html_tag }
```
Comment (andrewgarner) [inline config/application.rb]: Is it safe to remove this?

File: config/environments/production.rb [hunk 4]
```diff
@@ -53,16 +47,19 @@
   config.log_level = :debug
 
   # Prepend all log lines with the following tags.
-  # config.log_tags = [ :subdomain, :uuid ]
-
-  # Use a different logger for distributed setups.
-  # config.logger = ActiveSupport::TaggedLogging.new(SyslogLogger.new)
+  # config.log_tags = [ :request_id ]
```
Comment (andrewgarner) [inline config/environments/production.rb]: We've not commented this out in [previous PRs](https://github.com/alphagov/policy-publisher/pull/162/files#diff-1d98f0039bbc6df92adee882ed99d993R50)

File: config/initializers/new_framework_defaults.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file contains migration options to ease your Rails 5.0 upgrade.
+#
+# Once upgraded flip defaults one by one to migrate to the new default.
+#
+# Read the Guide for Upgrading Ruby on Rails for more info on each option.
+
+# Enable per-form CSRF tokens. Previous versions had false.
+Rails.application.config.action_controller.per_form_csrf_tokens = false
```
Comment (andrewgarner) [inline config/initializers/new_framework_defaults.rb]: Have you tried commenting each of these out and running the tests? If they pass we can remove this file and go with the new behaviour.

File: config/spring.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+%w(
+  .ruby-version
+  .rbenv-vars
+  tmp/restart.txt
+  tmp/caching-dev.txt
+).each { |path| Spring.watch(path) }
```
Comment (andrewgarner) [inline config/spring.rb]: Are we using Spring?

File: config/environments/production.rb (hunk 4 above)
Comment (whoojemaflip) [inline config/environments/production.rb]: It was [already commented out](https://github.com/alphagov/content-tagger/pull/345/files#diff-1d98f0039bbc6df92adee882ed99d993L56)

File: config/initializers/new_framework_defaults.rb (hunk 5 above)
Comment (whoojemaflip) [inline config/initializers/new_framework_defaults.rb]: Good thinking.

File: config/application.rb (hunk 3 above)
Comment (whoojemaflip) [inline config/application.rb]: It's all handled elsewhere:
- that `field_error_proc` has been moved to an [initializer](https://github.com/alphagov/content-tagger/pull/345/files#diff-a5458114f5f97709241748489a9041a2)
- the `Services` module has been moved to `app/lib`, meaning we don't need `/lib` on the eager_load_path, and
- `config.active_record.raise_in_transactional_callbacks = true` is deprecated in Rails 5

Comment (andrewgarner) [inline config/application.rb]: 👍

File: config/environments/production.rb (hunk 4 above)
Comment (andrewgarner) [inline config/environments/production.rb]: It was commented out with different example arguments `[ :subdomain, :uuid ]`.

Comment (whoojemaflip) [inline config/environments/production.rb]: I'd rather that either we delete the commentary entirely to leave just the config, or we leave the default Rails commentary.  Having custom commented out commentary seems a little pointless.

**Pull request merged**